client/camera/detectPrograms.js

```diff
@@ -398,6 +403,19 @@ export default function detectPrograms({ config, videoCapture, dataToRemember, d
     }
   });
 
+  // all points which haven't been matched to a shape are added as markers to the paper data
+  const markers = keyPoints
+    .filter(({ matchedShape }) => !matchedShape)
+    .map(({ colorIndex, avgColor, pt, size }) => {
+      const { x, y } = projectPointToUnitSquare(pt, videoMat, config.knobPoints);
+      return { size, position: { x, y }, avgColor, colorIndex };
+    });
+
+  // add markers to programs
+  programsToRender.forEach(programToRender => {
+    programToRender.markers = markers;
+  });
```

Comment:
Exposing the markers as data on every paper seems a bit strange to me. Maybe allow access to them via `paper.get('markers');`? Or only expose the markers that are contained within a paper on the paper (similarly to how it works on DL)?

Reply:
Maybe having both would be valuable?

- `myPaper.markers` exposes markers on `myPaper`
- `paper.get('markers')` exposes all markers that are _not_ on any piece of paper.

Reply:
I think the most flexible way would be to receive all markers with `paper.get('markers')`. I thought about having both APIs like @sgwilym suggested, but then it get's complicated for some case for example If you want to get all markers. Instead we could add a `paperId` property and `positionOnPaper` property if the marker is on a paper. 

Marker Object:

```JavaScript
{ 
  position: {x, y},  // global position
  color: [r, g, b]
  colorName // "red", "green", "blue" or "black"

  // these properties will be null if the marker is not on a paper
  paperId,
  positionOnPaper: {x, y} // position converted to coordinate system of the paper
}
```

Reply:
I like that idea a lot!